neurom/features/morphology.py

```diff
@@ -499,13 +541,28 @@ def _count_crossings(neurite, radius):
             center = morph.soma.center
         if radii is None:
             radii = [morph.soma.radius]
-    return [sum(_count_crossings(neurite, r)
-                for neurite in iter_neurites(morph, filt=is_type(neurite_type)))
-            for r in radii]
+
+    if isinstance(morph, Iterable):
+        sections = filter(is_type(neurite_type), morph)
+    else:
+        if use_subtrees:
+            sections = iter_sections(morph, section_filter=is_type(neurite_type))
```

Comment:
It's only these kinds of calls that make me wonder if the `section_filter` can be folded into the `neurite_type` filter, which is what I've been waffling about.  There's something about this if/else...

Reply:
The module is not formatted with black in general. I was thinking of making a pass on the entire package when v4 is merged so that we don't have a lot of irrelevant changes here.

Reply:
Well, for some reason this specific feature should also work with a list of sections...

from the docstring:
```
    This function can also be used with a list of sections, as follow::
        secs = (sec for sec in nm.iter_sections(morph) if complex_filter(sec))
        sholl = nm.features.neuritefunc.sholl_crossings(secs,
                                                        center=morph.soma.center,
                                                        radii=np.arange(0, 1000, 100))
```
I don't know why it is like this, but I didn't want to break backward compatibility even though it feels wrong to me to support sections at the morphology level of this feature.

Reply:
if we drop that, do any of the tests fail?  I'd consider removing it if not?

Reply:
Yeah, there is a test that checks the result of passing sections..

Reply:
nope, just the single test where a list of sections is passed.